scripts/dev/verify_idfs_in_cmake.py

```diff
@@ -69,31 +69,36 @@
     contents = f.read()
     matches = re.findall(r'\(([^)]+)\)', contents, re.MULTILINE)
     for match in matches:
-        if 'IDF_FILE' not in match:
-            continue
-        cleaned_match = match.replace('\n', '')
-        tokens = cleaned_match.split()  # special case that allows multiple whitespace delimiters
-        filename = tokens[1]
-        cmake_list_idf_files.add(filename)
+        if 'IDF_FILE' in match:
+            cleaned_match = match.replace('\n', '')
+            tokens = cleaned_match.split()  # special case that allows multiple whitespace delimiters
+            filename = tokens[1]
+            cmake_list_idf_files.add(filename)
+        elif 'PYTHON_FILE' in match:  # API-based IDF runs
+            cleaned_match = match.replace('\n', '')
+            tokens = cleaned_match.split()  # special case that allows multiple whitespace delimiters
+            filename = tokens[1]
+            filename = 'API/' + filename.replace('.py', '.idf')  # assuming the file is named the same as the py file
+            cmake_list_idf_files.add(filename)
```

Comment:
The new IDF in testfiles/API would cause a custom check failure because it wasn't listed the same way in the CMakeLists file.  So I modified this to also find matching lines for the new Python API test calls.